Seeding database fails

If I try to seed a database it fails. How to reproduce:

* `docker-compose down` # make sure none of the related containers are running
* `docker system prune -a` # remove all stopped containers, images and networks not in use
* `docker-compose run --rm freecodecamp npm install`#  install deps
* `docker-compose run --rm freecodecamp npm run only-once` # seed

Of interest is that ` entities not found, db may not be properly seeded` is ironic when you try to seed a db 😉 

Related but not identical #17256 -which shows a timeout connecting to the db, while my setup connects to the db just fine (`fcc:server db connected +0ms`).

```sh
▶ docker-compose run --rm freecodecamp npm run only-once
Starting freecodecamp_mailhog_1 ... done
Starting freecodecamp_db_1      ... done

> freecodecamp@0.1.0 only-once /app
> npm run prelint-js && echo '/****/' && echo 'Seeding Database' && echo '/****/' && node seed/index.js && echo '/****/' && echo 'Seeding Completed' && echo '/****/'


> freecodecamp@0.1.0 prelint-js /app
> npm run ensure-env


> freecodecamp@0.1.0 ensure-env /app
> node ./config/ensure-env.js



rev-manifest present




pathMigration present


/****/
Seeding Database
/****/
Warning: Accessing createClass via the main React package is deprecated, and will be removed in React v16.0. Use a plain JavaScript class instead. If you're not yet ready to migrate, create-react-class v15.* is available on npm as a temporary, drop-in replacement. For more info see https://fb.me/react-create-class
Warning: Accessing PropTypes via the main React package is deprecated, and will be removed in  React v16.0. Use the latest available v15.* prop-types package from npm instead. For info on usage, compatibility, migration and more, see https://fb.me/prop-types-docs
  fcc:server db connected +0ms
/app/node_modules/loopback-connector-mongodb/node_modules/mongodb/lib/utils.js:123
    process.nextTick(function() { throw err; });
                                  ^

Error: Error: entities not found, db may not be properly seeded
    at AnonymousObserver._onError (/app/server/boot/home.js:18:21)
    at AnonymousObserver.Rx.AnonymousObserver.AnonymousObserver.error (/app/node_modules/rx/dist/rx.js:1836:12)
    at AnonymousObserver.Rx.internals.AbstractObserver.AbstractObserver.onError (/app/node_modules/rx/dist/rx.js:1772:14)
    at AnonymousObserver.o.onError (/app/node_modules/rx/dist/rx.js:1999:22)
    at AnonymousObserver.tryCatcher (/app/node_modules/rx/dist/rx.js:63:31)
    at AutoDetachObserverPrototype.error (/app/node_modules/rx/dist/rx.js:5891:52)
    at AutoDetachObserver.Rx.internals.AbstractObserver.AbstractObserver.onError (/app/node_modules/rx/dist/rx.js:1772:14)
    at InnerObserver.error (/app/node_modules/rx/dist/rx.js:4530:14)
    at InnerObserver.Rx.internals.AbstractObserver.AbstractObserver.onError (/app/node_modules/rx/dist/rx.js:1772:14)
    at InnerObserver.o.onError (/app/node_modules/rx/dist/rx.js:1999:22)
    at InnerObserver.tryCatcher (/app/node_modules/rx/dist/rx.js:63:31)
    at AutoDetachObserverPrototype.error (/app/node_modules/rx/dist/rx.js:5891:52)
    at AutoDetachObserver.Rx.internals.AbstractObserver.AbstractObserver.onError (/app/node_modules/rx/dist/rx.js:1772:14)
    at AutoDetachObserver.o.onError (/app/node_modules/rx/dist/rx.js:1999:22)
    at AutoDetachObserver.o.onError (/app/node_modules/rx/dist/rx.js:1999:22)
    at /app/node_modules/rx/dist/rx.js:1904:72
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! freecodecamp@0.1.0 only-once: `npm run prelint-js && echo '/****/' && echo 'Seeding Database' && echo '/****/' && node seed/index.js && echo '/****/' && echo 'Seeding Completed' && echo '/****/'`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the freecodecamp@0.1.0 only-once script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2018-05-29T00_05_10_316Z-debug.log
```

--- a/package.json
+++ b/package.json
@@ -17,7 +17,7 @@
     "lint-utils": "jsonlint-cli server/utils/*.json",
     "lint-js": "eslint --ext=.js,.jsx gulpfile.js server/ common/ config/ client/",
     "lint-json": "npm run lint-server && npm run lint-challenges && npm run lint-resources && npm run lint-utils",
-    "only-once": "npm run prelint-js && echo '/****/' && echo 'Seeding Database' && echo '/****/' && node seed/index.js && echo '/****/' && echo 'Seeding Completed' && echo '/****/'",
+    "only-once": "npm run prelint-js && echo '/****/' && echo 'Seeding Database' && echo '/****/' && SEEDING=true node seed/index.js && echo '/****/' && echo 'Seeding Completed' && echo '/****/'",
     "postonly-once": "gulp generate-migration-map",
     "postseed": "gulp generate-migration-map",
     "prelint-js": "npm run ensure-env",

--- a/server/boot/home.js
+++ b/server/boot/home.js
@@ -9,19 +9,22 @@ module.exports = function(app, done) {
   const { About } = app.models;
   const router = app.loopback.Router();
   let challengeCount = 0;
-  cachedMap(app.models)
-    .do(({ entities: { challenge } }) => {
-      challengeCount = Object.keys(challenge).length;
-    })
-    .subscribe(
-      () => {},
-      err => {throw new Error(err);},
-      () => {
-        router.get('/', addDefaultImage, index);
-        app.use(router);
-        done();
-      }
-    );
+
+  if (!process.env.SEEDING) {
+    cachedMap(app.models)
+      .do(({ entities: { challenge } }) => {
+        challengeCount = Object.keys(challenge).length;
+      })
+      .subscribe(
+        () => {},
+        err => {throw new Error(err);},
+        () => {
+          router.get('/', addDefaultImage, index);
+          app.use(router);
+          done();
+        }
+      );
+  }
 
   function addDefaultImage(req, res, next) {
     if (!req.user || req.user.picture) {

